mlflow/deployments/__init__.py

```diff
@@ -0,0 +1,24 @@
+from mlflow.deployments.base_plugin import BasePlugin
+from mlflow.deployments import interface
+
+
+# TODO: It's a good practise to avoid using ``list`` here
+
+__all__ = ['BasePlugin', 'create', 'delete', 'update', 'list', 'describe']
+
+
+def __dir__():
```

Comment:
Hm, what's the purpose of defining `__dir__` and `__getattr__` here? Could we perform the lazy loading in interface.py itself?

Reply:
I don't think we can. Since those functions are defined in interface.py, the name scope of that file gets precedence over the `__getattr__`, so the function call won't reach `__getattr__`.
Even if we find a way to do that, we'll end up doing the plugin registration inside each interface function. We probably don't want that. 
Why do you prefer not doing this in `__init__.py`?

Reply:
Got it, mainly just didn't quite understand what this code was doing / it adds some complexity. Would you mind adding a clarifying comment to the code explaining your reasoning above?

Reply:
Will do